src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.item = element;
+            this.next = next;
+            this.prev = prev;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> node = new Node<>(tail, value, null);
+        if (head == null) {
+            head = node;
+        } else {
+            tail.next = node;
+        }
+        tail = node;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkIndex(index, size);
+        if (index == size) {
+            add(value);
+        } else {
+            Node<T> nodeUnderIndex = findNode(index);
+            Node<T> newNode = new Node<>(nodeUnderIndex.prev, value, nodeUnderIndex);
+            nodeUnderIndex.prev = newNode;
+            if (newNode.prev == null) {
+                head = newNode;
+            } else {
+                newNode.prev.next = newNode;
+            }
+            size++;
```

Comment:
This code can be moved to another function as well

Reply:
done